File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size())
            return false;
        
        std::array<int, 26> dict = {0};

        for (std::size_t idx = 0; idx < s.size(); ++idx) {
            ++dict[s[idx] - 'a'];
            --dict[t[idx] - 'a'];
        }

        for (std::size_t idx = 0; idx < dict.size(); ++idx)
            if (dict[idx] != 0)
                return false;

        return true;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
from collections import defaultdict
import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freqs = defaultdict(int)
        for num in nums:
            freqs[num] += 1
        
        buckets = [[] for _ in range(len(nums) + 1)]
        print(buckets)
        print(freqs)
        for num, freq in freqs.items():
            print(num, freq)
            buckets[freq].append(num)
        
        print(buckets)
        res = []
        for idx in range(len(nums), -1, -1):
            for num in buckets[idx]:
                res.append(num)
                if (len(res) == k):
                    return res